Missing `-ldl` for examples

Docker image fails to build:

```
[ 35%] Building CXX object examples/cpp_api/CMakeFiles/object_cpp.dir/object.cc.o
[ 36%] Linking CXX executable object_cpp
../../tiledb/libtiledb.so: undefined reference to `dladdr'
../../tiledb/libtiledb.so: undefined reference to `dlopen'
../../tiledb/libtiledb.so: undefined reference to `dlclose'
../../tiledb/libtiledb.so: undefined reference to `dlerror'
../../tiledb/libtiledb.so: undefined reference to `dlsym'
collect2: error: ld returned 1 exit status
```

--- a/examples/c_api/CMakeLists.txt
+++ b/examples/c_api/CMakeLists.txt
@@ -36,15 +36,15 @@ add_custom_target(examples_c DEPENDS tiledb_shared)
 # Function that builds an executable per example
 function(build_TileDB_example_capi TARGET)
   add_executable(${TARGET}_c EXCLUDE_FROM_ALL ${TARGET}.c)
-  target_link_libraries(${TARGET}_c tiledb_shared)
-  if (NOT WIN32)
-    # On Linux, must explicitly link -lpthread in order for static linking
-    # to libzstd or libcurl to work.
-    target_link_libraries(${TARGET}_c pthread)
-  endif()
   target_include_directories(${TARGET}_c PUBLIC
     "${CMAKE_CURRENT_BINARY_DIR}/../"
   )
+  target_link_libraries(${TARGET}_c tiledb_shared)
+  if (NOT WIN32)
+    # On Linux, must explicitly link -lpthread -ldl in order for static linking
+    # to libzstd or libcurl to work.
+    target_link_libraries(${TARGET}_c pthread dl)
+  endif()
   add_dependencies(examples_c ${TARGET}_c)
 endfunction()
 

--- a/examples/cpp_api/CMakeLists.txt
+++ b/examples/cpp_api/CMakeLists.txt
@@ -31,10 +31,15 @@ add_custom_target(examples_cpp DEPENDS tiledb_shared)
 # Function that builds an executable per example
 function(build_TileDB_example_cppapi TARGET)
   add_executable(${TARGET}_cpp EXCLUDE_FROM_ALL ${TARGET}.cc)
-  target_link_libraries(${TARGET}_cpp tiledb_shared)
   target_include_directories(${TARGET}_cpp PUBLIC
     "${CMAKE_CURRENT_BINARY_DIR}/../"
   )
+  target_link_libraries(${TARGET}_cpp tiledb_shared)
+  if (NOT WIN32)
+    # On Linux, must explicitly link -lpthread -ldl in order for static linking
+    # to libzstd or libcurl to work.
+    target_link_libraries(${TARGET}_cpp pthread dl)
+  endif()
   add_dependencies(examples_cpp ${TARGET}_cpp)
 endfunction()
 

